lib/features/mfa/bloc/mfa_bloc.dart

```diff
@@ -107,29 +102,17 @@ class MfaBloc extends Bloc<MfaEvent, MfaState> {
           ),
         );
       }
-    } on KeycloakClientException catch (err) {
-      if (err.type == KeycloakExceptionType.notRegistered) {
+    } on KeycloakClientException catch (error) {
+      if (error.type == KeycloakExceptionType.notRegistered) {
         try {
           await _service.delete(_authenticator!);
           _authenticator = null;
           emit(state.copyWith(status: MfaStatus.setup));
         } catch (_) {}
       }
-      emit(
-        state.copyWith(
-          error: err.toString(),
-          loginAttempt: null,
-          isLoading: false,
-        ),
-      );
-    } catch (err) {
-      emit(
-        state.copyWith(
-          error: err.toString(),
-          loginAttempt: null,
-          isLoading: false,
-        ),
-      );
+      emit(state.copyWith(error: event.raiseError ? error : null, loginAttempt: null, isLoading: false));
+    } catch (error) {
+      emit(state.copyWith(error: event.raiseError ? error : null, loginAttempt: null, isLoading: false));
```

Comment:
Mind the `event.raiseError` check. I am not super happy with this solution but I think its okay.